compilers/concrete-compiler/compiler/lib/Common/Keysets.cpp

```diff
@@ -0,0 +1,324 @@
+// Part of the Concrete Compiler Project, under the BSD3 License with Zama
+// Exceptions. See
+// https://github.com/zama-ai/concrete-compiler-internal/blob/main/LICENSE.txt
+// for license information.
+
+#include "concretelang/Common/Keysets.h"
+#include "capnp/message.h"
+#include "concrete-cpu.h"
+#include "concrete-protocol.capnp.h"
+#include "concretelang/Common/Csprng.h"
+#include "concretelang/Common/Error.h"
+#include "concretelang/Common/Keys.h"
+#include "kj/common.h"
+#include "kj/io.h"
+#include "llvm/ADT/ScopeExit.h"
+#include "llvm/Support/FileSystem.h"
+#include "llvm/Support/Path.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <iostream>
+#include <stdlib.h>
+#include <string>
+#include <unistd.h>
+#include <utime.h>
+
+using concretelang::csprng::ConcreteCSPRNG;
+using concretelang::error::Result;
+using concretelang::error::StringError;
+using concretelang::keys::LweBootstrapKey;
+using concretelang::keys::LweKeyswitchKey;
+using concretelang::keys::LweSecretKey;
+using concretelang::keys::PackingKeyswitchKey;
+
+/// The default reading limit of capnp must be increased for large keys.
+const capnp::ReaderOptions KEY_READER_OPTS =
+    capnp::ReaderOptions{7000000000, 64};
+
+#ifdef CONCRETELANG_GENERATE_UNSECURE_SECRET_KEYS
+inline void getApproval() {
```

Comment:
This could be factored out from `Keysets.cpp` and `Keys.cpp` into a header file as `static inline void getApproval()`.

Reply:
Done thanks !